react-frontend/src/pages/ListingPage.js

```diff
@@ -1,85 +1,108 @@
 import React, { useEffect, useState, useContext } from 'react'
-import { useParams, useNavigate } from 'react-router-dom'
+import { useParams } from 'react-router-dom'
 import { AuthContext } from '../../context/AuthContext'
-import Header from '../components/Header/Header'
-import Nav from '../components/Navigation/Nav'
+import Layout from '../components/Layout/Layout'
 import TextInput from '../components/TextInput/TextInput'
 import Button from '../components/Button/Button'
 import Popup from '../components/Popup/Popup'
 import Listing from '../components/Listing/Listing'
 import PreviousBids from '../components/PreviousBids/PreviousBids'
 import DetailedDescription from '../components/DetailedDescription/DetailedDescription'
-import { getListingById, placeBid, addToWatchlist, getBidsByListingId } from '../api'
+import Alert from '../components/Alert/Alert'
+import {
+  getListingById,
+  placeBid,
+  getBidsByListingId,
+  closeListing,
+  fetchWatchlistIds
+} from '../api'
 import './styles/ListingPage.css'
 
 const ListingPage = () => {
   const { listingId } = useParams()
-  const navigate = useNavigate()
-  const { isAuthenticated } = useContext(AuthContext)
+  const { isAuthenticated, user } = useContext(AuthContext)
   const [listing, setListing] = useState(null)
   const [bidAmount, setBidAmount] = useState('')
   const [isPopupOpen, setIsPopupOpen] = useState(false)
   const [isSuccess, setIsSuccess] = useState(false)
   const [errorMessage, setErrorMessage] = useState('')
   const [bids, setBids] = useState([])
+  const [showAlert, setShowAlert] = useState(false)
+  const [watchlist, setWatchlist] = useState([])
+  const [isInWatchlist, setIsInWatchlist] = useState(false)
 
   useEffect(() => {
-    const fetchListing = async () => {
+    const fetchListingAndBids = async () => {
       try {
-        const data = await getListingById(listingId)
-        setListing(data)
+        const listingData = await getListingById(listingId)
+        setListing(listingData)
+        const bidsData = await getBidsByListingId(listingId)
+        setBids(bidsData)
       } catch (error) {
-        console.error('Error fetching listing:', error)
+        console.error('Error fetching listing or bids:', error)
```

Comment:
Let's handle error state

Reply:
Added error state in this commint: [1745875](https://github.com/YaroslavKSE/AuctionWeb/pull/3/commits/174587596ab873a6ae4b2fbf2caf2a3cd19a0569)